test/Claims.t.sol

```diff
@@ -0,0 +1,91 @@
+
+    function setUp() public {}
+
+    function testCanBurn(uint256 amount) public {
+        assertEq(claimsImpl.balanceOf(address(this), currency0), 0);
+        vm.expectEmit(true, true, false, false);
+        emit Mint(address(this), currency0, amount);
+        claimsImpl.mint(address(this), currency0, amount);
+        assertEq(claimsImpl.balanceOf(address(this), currency0), amount);
+
+        vm.expectEmit(true, true, false, false);
+        emit Burn(address(this), currency0, amount);
+        claimsImpl.burn(currency0, amount);
+        assertEq(claimsImpl.balanceOf(address(this), currency0), 0);
+    }
+
+    function testCatchesUnderflowOnBurn(uint256 amount) public {
+        vm.assume(amount < type(uint256).max - 1);
+        vm.expectEmit(true, true, false, false);
+        emit Mint(address(this), currency0, amount);
+        claimsImpl.mint(address(this), currency0, amount);
+
+        assertEq(claimsImpl.balanceOf(address(this), currency0), amount);
+        vm.expectRevert(abi.encodeWithSelector(IClaims.InsufficientBalance.selector));
+        claimsImpl.burn(currency0, amount + 1);
+    }
+
+    function testCanTransfer(uint256 amount) public {
+        vm.expectEmit(true, true, false, false);
+        emit Mint(address(this), currency0, amount);
+        claimsImpl.mint(address(this), currency0, amount);
+
+        assertEq(claimsImpl.balanceOf(address(this), currency0), amount);
+        vm.expectEmit(true, true, true, false);
+        emit Transfer(address(this), address(1), currency0, amount);
+        claimsImpl.transfer(address(1), currency0, amount);
+        assertEq(claimsImpl.balanceOf(address(this), currency0), 0);
+        assertEq(claimsImpl.balanceOf(address(1), currency0), amount);
+    }
+
+    function testCatchesUnderflowOnTransfer(uint256 amount) public {
+        vm.assume(amount < type(uint256).max - 1);
+        vm.expectEmit(true, true, false, false);
+        emit Mint(address(this), currency0, amount);
+        claimsImpl.mint(address(this), currency0, amount);
+
+        assertEq(claimsImpl.balanceOf(address(this), currency0), amount);
+        vm.expectRevert(abi.encodeWithSelector(IClaims.InsufficientBalance.selector));
+        claimsImpl.transfer(address(1), currency0, amount + 1);
+    }
+
+    function testCatchesOverflowOnTransfer() public {
+        claimsImpl.mint(address(0xdead), currency0, type(uint256).max);
+        claimsImpl.mint(address(this), currency0, 1);
+        // transfer will revert since overflow
+        vm.expectRevert();
+        claimsImpl.transfer(address(0xdead), currency0, 1);
+    }
+
+    function testCanTransferToZeroAddress() public {
+        claimsImpl.mint(address(this), currency0, 1);
+        vm.expectEmit(true, true, true, false);
+        emit Transfer(address(this), address(0), currency0, 1);
+        claimsImpl.transfer(address(0), currency0, 1);
+        assertEq(claimsImpl.balanceOf(address(this), currency0), 0);
```

Comment:
should assertEq also of:
- address(this) balance before the transfer
- address(0) balance before the transfer
- address(0) balance after the transfer

Reply:
just added!